batch2/batch/driver/driver.py

```diff
@@ -82,26 +89,49 @@ def __init__(self, driver, name, spec, output_directory, cores_mcpu, instance=No
         self.lock = asyncio.Lock(loop=loop)
 
     async def config(self):
-        future_secrets = []
-        secret_names = []
-
-        for volume in self.spec['spec']['volumes']:
-            if volume['secret'] is not None:
-                name = volume['secret']['secret_name']
-                secret_names.append(name)
-                future_secrets.append(self.driver.k8s.read_secret(name))
-        results = await asyncio.gather(*future_secrets)
-
-        secrets = {}
-        for name, (secret, err) in zip(secret_names, results):
-            if err is not None:
+        job_spec = dict(self.job_spec)
+
+        # copy secrets
+        if 'secrets' in job_spec:
+            job_spec['secrets'] = [dict(secret) for secret in job_spec['secrets']]
+
+        # update resources with defaults
+        if 'resources' not in job_spec:
+            job_spec['resources'] = {}
+        resources = job_spec['resources']
+        if 'cpu' not in resources:
+            resources['cpu'] = BATCH_JOB_DEFAULT_CPU
+        if 'memory' not in resources:
+            resources['memory'] = BATCH_JOB_DEFAULT_CPU
+
+        # add user's gsa-key
+        secrets = job_spec.get('secrets', [])
+        secrets = list(secrets)
+        secrets.append({
+            'namespace': 'batch-pods',  # FIXME unused
+            'name': self.userdata['gsa_key_secret_name'],
+            'mount_path': '/gsa-key',
+            'mount_in_copy': True
+        })
+        job_spec['secrets'] = secrets
+
+        secret_futures = []
+        for secret in secrets:
+            secret_futures.append(self.driver.k8s.read_secret(secret['name']))
```

Comment:
Can you add a FIXME or something to know that we need to check the user should have access to that secret?